### PR TITLE
Mohieddin

### PR DESCRIPTION
**Pull request by Md-Almasri**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -10,6 +10,19 @@ app.set('view engine', 'hbs');
 app.use(express.static(path.join(__dirname, 'public')));
 
 // add your routes here :)
+app.get('/:code', function (req,res) {
+	const url = 'http://localhost:8080/code/' + req.params.code;
+	const url2 = 'http://numbersapi.com/' + req.params.code;
+	const promise1 = fetch(url).then((res) => res.json());
+	const promise2 = fetch(url2).then((res) => res.text());
```
Comment (drainpip) [inline server.js]: I'm not a fan of making variables so generic, when you come across them later after their initialization you might have forgotten their context. Something like `localhostUrl` and `numbersApiUrl` (and then maybe `localhostUrlPromise`) would make them easy to keep track of. It's no big deal in production as you should minify/uglify everything so the bytes down the line won't be a factor :)

Comment (Md-Almasri) [reply server.js]: Thank you very much

File: server.js [hunk 2]
```diff
@@ -11,6 +11,55 @@ app.use(express.static(path.join(__dirname, 'public')));
...
+});
+
+app.get('/', (req, res) => {
+  const localhostUrl = 'http://localhost:8080/code';
+  fetch(localhostUrl, {
+    method: 'GET',
+  })
+  .then((apiRes) => {
+    apiRes.json()
+      .then((json) => {
```
Comment (TheoLeanse) [inline server.js]:
```
functionWhichReturnsAPromise()
  .then(result => {
    return result.methodThatReturnsAPromise()
      .then(result => {
        return doSomething(result);
      });
  });
```
is the same as...
```
functionWhichReturnsAPromise()
  .then(result => {
    return result.methodWhichReturnsAPromise();
  })
  .then(result => {
    return doSomething(result);
  })
```

Comment (Md-Almasri) [reply server.js]: Ah Ok 
Thanks @TheoLeanse

Comment (TheoLeanse) [reply server.js]: 👌 perfect, nice one

**Comment by TheoLeanse** [comment]
Hope you're well!

**Comment by Md-Almasri** [comment]
Hi @TheoLeanse ,
I am good, How are you?

**Pull request closed without merging**